eisbuk-admin/src/store/selectors/customers.ts

```diff
@@ -22,3 +23,41 @@ export const getCustomersList = (state: LocalStore): Customer[] => {
   const customersInStore = getCustomersRecord(state);
   return Object.values(customersInStore);
 };
+
+/**
+ * Get a list of all the customers whose birthdays are today
+ * (not an actual firestore query but rather synced local store entry)
+ * @param state Local Redux Store
+ * @returns list of customers grouped by birthday
+ */
+export const getCustomersWithBirthday = (
+  state: LocalStore
+): CustomerBirthday => {
+  const customersInStore = getCustomersRecord(state);
+
+  // find index of birthday thats >= todays date to slice at that and concat the rest at the end
+  const sortedCustomers = Object.values(customersInStore).sort((a, b) =>
+    DateTime.fromISO(a.birthday)
+      .set({ year: 2000 })
+      .toString()
+      .localeCompare(
+        DateTime.fromISO(b.birthday).set({ year: 2000 }).toString()
+      )
+  );
+  const index = sortedCustomers.findIndex(
+    (c) =>
+      DateTime.now().startOf("day").set({ year: 2000 }) <=
```

Comment:
This function ideally should be pure, i.e. its result should only depend on the passed parameters. Striving to have as many pure functions as possible results in an easier to maintain codebase, since side effects are contained.

For this reason it would be better not to invoke `now` here, but rather rely on a date passed by the caller. The UI code that will use this function will be responsible of invoking `now()` and passing the information down here.